Apache-Arrow-Flight-Manager-Tester/main.py

```diff
@@ -0,0 +1,116 @@
+import pprint
+from typing import Literal
+
+from pyarrow import flight, Schema
+from pyarrow._flight import FlightInfo, FlightClient, ActionType, FlightEndpoint
+
+
+# PyArrow Functions.
+def list_flights(flight_client: FlightClient) -> list[FlightInfo]:
+    response = flight_client.list_flights()
+
+    return list(response)
+
+
+def get_schema(flight_client: FlightClient, table_name: str) -> Schema:
+    upload_descriptor = flight.FlightDescriptor.for_path(table_name)
+    response = flight_client.get_schema(upload_descriptor)
+
+    return response.schema
+
+
+def do_action(flight_client: FlightClient, action_type: str, action_body: str) -> list[any]:
+    action_body_bytes = str.encode(action_body)
+    action = flight.Action(action_type, action_body_bytes)
+    response = flight_client.do_action(action)
+
+    return list(response)
+
+
+def list_actions(flight_client: FlightClient) -> list[ActionType]:
+    response = flight_client.list_actions()
+
+    return list(response)
+
+
+# Helper functions.
+def list_table_names(flight_client: FlightClient) -> list[str]:
+    flights = list_flights(flight_client)
+
+    return [table_name.decode("utf-8") for table_name in flights[0].descriptor.path]
+
+
+def initialize_database(flight_client: FlightClient, tables: list[str]) -> list[str]:
+    result = do_action(flight_client, "InitializeDatabase", ",".join(tables))[0]
+    decoded_result = result.body.to_pybytes().decode("utf-8")
+
+    return decoded_result.split(";")
+
+
+def execute_query(flight_client: FlightClient, query: str) -> None:
+    # Retrieve the flight info that describes how to execute the query.
+    query_descriptor = flight.FlightDescriptor.for_command(query)
+    flight_info: FlightInfo = flight_client.get_flight_info(query_descriptor)
```

Comment:
Unclear why the type of `flight_info` and `endpoint` is specified by the type is not specified for the remaining types? Maybe add comment to document why.

Reply:
Types were added to variables in some places if the PyArrow types were not helpful (which they often are not). However, since this is just utility code I have removed it here for consistency.